tests/spec/date_series_ops/test_date_series_ops.py

```diff
@@ -132,19 +132,34 @@ def test_add_months(spec_test):
     #
     #       The smallest y such that `(y - x).months == N`
     #
-    #  2. It is what SQLite does.
+    #  2. It is what SQLite does *in most cases*.  SQLite rolls over to the first of the next
+    #     month, except in the case of the last days of February, where the rolled-over date
+    #     from 30/31 Feb becomes 02/03 March.
     #
     # Other databases take different approachs so we have to work around their behaviour
-    # in their respective query engines.
+    # in their respective query engines, and in the SQLite engine for the end of February
+    # behaviour.
     table_data = {
         p: """
-          |     d1     | i1
-        --+------------+-----
-        1 | 2003-01-29 |  1
-        2 | 2004-01-29 |  1
-        3 | 2003-01-29 | -1
-        4 | 2000-10-31 |  11
-        5 | 2000-10-31 | -11
+           |     d1     | i1
+        ---+------------+-----
+        1  | 2003-01-29 |  1
+        2  | 2004-01-29 |  1
+        3  | 2003-01-30 |  1
+        4  | 2004-01-30 |  1
+        5  | 2003-01-31 |  1
+        6  | 2004-01-31 |  1
+        7  | 2003-03-29 | -1
+        8  | 2004-03-29 | -1
+        9  | 2003-03-30 | -1
+        10 | 2004-03-30 | -1
+        11 | 2003-03-31 | -1
+        12 | 2004-03-31 | -1
+        13 | 2003-01-29 | -1
+        14 | 2000-10-31 |  11
+        15 | 2000-10-31 | -11
+        16 | 2000-10-31 |  4
+        17 | 1999-11-30 |  3
```

Comment:
These test cases may be overkill, especially given that it ends up in the docs, but I wanted to assure myself it worked.  I could remove some if it's going to make the docs confusing.

Reply:
Yeah, I can see why you'd add all these while working on it but I think we need to be trying to keep a lid on the spec tests otherwise it's very easy for them to get out of control. I think what we should be aiming for is the minimal set that clearly excludes SQLite's native behaviour.